static/js/components/VideoPlayer.js

```diff
@@ -1,37 +1,225 @@
+      video.videosubtitle_set.map((subtitle: VideoSubtitle) => {
+        const subUrl = makeVideoSubtitleUrl(subtitle);
+        if (!trackUrls.includes(subUrl)) {
+          this.player.addRemoteTextTrack({
+            kind: "captions",
+            src: subUrl,
+            srcLang: subtitle.language,
+            label: subtitle.language_name
+          });
+        }
+      });
+    }
+  }
+
+  drawCanvas(canvas: HTMLCanvasElement, shiftX: boolean, shiftY: boolean) {
+    function drawCanvasImage(canvas, videoNode, shiftX, shiftY) {
+      const x = shiftX ? Math.floor(videoNode.videoWidth / 2) : 0;
+      const y = shiftY ? Math.floor(videoNode.videoHeight / 2) : 0;
+      const context = canvas.getContext('2d');
+      context.drawImage(
+        videoNode,
+        x, y, Math.floor(videoNode.videoWidth / 2), Math.floor(videoNode.videoHeight / 2),
+        0, 0, canvas.width, canvas.height
+      );
+      setTimeout(drawCanvasImage, 20, canvas, videoNode, shiftX, shiftY);
+    }
+
+    canvas.width = Math.floor(this.videoNode.offsetWidth / 4) - 2;
+    canvas.height = Math.floor(this.videoNode.offsetHeight / 4) - 2;
+    if (canvas && this.videoNode) {
+      drawCanvasImage(canvas, this.videoNode, shiftX, shiftY);
+    }
+  }
+
+  configureCameras() {
+    if (this.cameras) {
+      const canvasElements = this.cameras.getElementsByTagName("canvas");
+      Object.keys(canvases).map(corner => {
+        this.drawCanvas(
+          canvasElements[corner],
+          canvases[corner].shiftX,
+          canvases[corner].shiftY
+        );
+      });
+    }
+  }
+
+  isFullscreen() {
+    // $FlowFixMe
+    return document[FULLSCREEN_API.fullscreenElement];
+  }
+
+  cropVideo() {
+    const { selectedCorner } = this.props;
+    if (selectedCorner) {
+      const shiftX = canvases[selectedCorner].shiftX;
+      const shiftY = canvases[selectedCorner].shiftY;
+      const transformProps = ['transform', 'WebkitTransform', 'MozTransform', 'msTransform', 'OTransform'];
+      let prop = transformProps[0];
+
+      transformProps.some((property: string) => {
+        prop = property;
+        return (typeof this.player.el_.style[prop] !== 'undefined');
+      });
+
+      const aspectRatio = (this.player.videoWidth() / this.player.videoHeight());
+      let videoWidth = Math.min(
+        parseInt(window.getComputedStyle(this.videoNode).maxHeight) * aspectRatio,
+        window.innerWidth,
+        screen.width
+      );
+      if (isNaN(videoWidth) || this.isFullscreen()) {
+        videoWidth = Math.min(window.innerWidth, screen.width);
+      }
+      const canvasWidth = Math.floor(videoWidth / 4);
+      videoWidth = Math.floor(videoWidth - (canvasWidth - (canvasWidth / aspectRatio / 3)));
+      this.videoContainer.style.maxWidth = `${videoWidth}px`;
+      this.videoContainer.parentElement.style.width = `${videoWidth + canvasWidth}px`;
+
+      const left = Math.round(this.player.currentWidth() / (shiftX ? -2 : 2));
```

Comment:
Can you explain this a bit? I'm not sure what the `2` values are for here, or the `scale(2)` below

Reply:
`scale(2)` is what causes the main video screen to zoom in to a corner. The 2 or -2 for `left`, `top` determines in which direction the left and top origins should be shifted to display the correct corner.

Reply:
Can you add this as a comment in the code?